File: algo_go/callatz.go
package main

import (
    "fmt"
)

func main() {
   var n, step int;
   fmt.Scanf("%d", &n);

   for ; (n != 1); {
       if n % 2 == 0 {
           n = n / 2;
       } else {
           n = (3 * n + 1) / 2
       }
       step++;
   }
   fmt.Printf("%d\n", step);
}

File: algo-c/oldfiles/callatz.c
#include <stdio.h>

int main(void) {
    int n, step = 0;
    scanf("%d", &n);
    while (n != 1) {
        if (n % 2 == 0)
            n = n / 2;
        else
            n = (3 * n + 1) / 2;
        step = step + 1;
    }
    printf("%d\n", step);
    return 0;
}